gen/aircraft.py

```diff
@@ -1158,17 +1165,34 @@ def build(self) -> MovingPoint:
         self.set_waypoint_tot(waypoint, self.timing.ingress)
         return waypoint
 
-class CasIngressBuilder(PydcsWaypointBuilder):
+class CasIngressBuilder(IngressBuilder):
     def build(self) -> MovingPoint:
         waypoint = super().build()
-        self.set_waypoint_tot(waypoint, self.timing.ingress)
-        waypoint.add_task(EngageTargets(max_distance=nm_to_meter(10),
+        ingress_waypoint = self.get_flight_point('INGRESS')
+        cas_waypoint = self.get_flight_point('CAS')
+        try:
+            ingress_point = Point(ingress_waypoint.position.x, ingress_waypoint.position.y)
+        
+            waypoint.add_task(EngageTargetsInZone(
+                                position=cas_waypoint,
+                                radius=ingress_point.distance_to_point(
+                                    cas_waypoint.position
+                                    ),
+                                targets=[
+                                    Targets.All.GroundUnits.GroundVehicles,
+                                    Targets.All.GroundUnits.AirDefence.AAA,
+                                    Targets.All.GroundUnits.Infantry,
+                                ])
+            )
+        except AttributeError:
+            logging.error(f'Unable to create CAS target zone.  Falling back to search and engage')
```

Comment:
`logging.exception` makes this more debuggable.

Reply:
Updated in fa01303